rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -61,217 +72,564 @@ class LiaisonNegotiator : public rmf_traffic::schedule::Negotiator
-  if (n == 0)
-    return rmf_utils::nullopt;
+    return;
+  }
 
+  // Determine task type and convert to request pointer
+  rmf_task::ConstRequestPtr new_request = nullptr;
+  const auto& task_profile = msg->task_profile;
+  const auto& task_type = task_profile.task_type;
+  const rmf_traffic::Time start_time = rmf_traffic_ros2::convert(task_profile.start_time);
+  // TODO (YV) get rid of ID field in RequestPtr
+  std::string id = msg->task_profile.task_id;
   const auto& graph = planner->get_configuration().graph();
-  const auto loop_start_wp = graph.find_waypoint(request.start_name);
-  if (!loop_start_wp)
-    return rmf_utils::nullopt;
 
-  const auto loop_end_wp = graph.find_waypoint(request.finish_name);
-  if (!loop_end_wp)
-    return rmf_utils::nullopt;
+  // Process Cleaning task
+  if (task_type.type == rmf_task_msgs::msg::TaskType::TYPE_CLEAN)
+  {
+    if (task_profile.clean.start_waypoint.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [clean.start_waypoint] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    // Check for valid start waypoint
+    const std::string start_wp_name = task_profile.clean.start_waypoint;
+    const auto start_wp = graph.find_waypoint(start_wp_name);
+    if (!start_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), start_wp_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    // Get dock parameters
+    const auto clean_param_it = dock_param_map.find(start_wp_name);
+    if (clean_param_it == dock_param_map.end())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Dock param for dock_name:[%s] unavailable. Rejecting BidNotice with "
+        "task_id:[%s]", start_wp_name.c_str(), id.c_str());
+
+      return;
+    }
+    const auto clean_param = clean_param_it->second;
+
+    // Check for valid finish waypoint
+    const std::string finish_wp_name = clean_param.finish;
+    const auto finish_wp = graph.find_waypoint(finish_wp_name);
+    if (!finish_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), finish_wp_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    // Interpolate docking waypoint into trajectory
+    std::vector<Eigen::Vector3d>  positions;
+    for (const auto& location: clean_param.path)
+      positions.push_back({location.x, location.y, location.yaw});
+    rmf_traffic::Trajectory cleaning_trajectory =
+      rmf_traffic::agv::Interpolate::positions(
+        planner->get_configuration().vehicle_traits(),
+        start_time,
+        positions);
+    
+    if (!(cleaning_trajectory.size() > 0))
```

Comment:
If you merge in the latest `master`, you can change this to `if (cleaning_trajectory.empty())`. Or even without the latest `master`, this could be `if (cleaning_trajectory.size() == 0)`

Reply:
Done https://github.com/osrf/rmf_core/pull/217/commits/f73f6896aedf9ba33247fe43c9034138912a848a